### PR TITLE
Scorecard refactor run

### PR DESCRIPTION
**Pull request by jmccormick2001**
**Description of the change:**
this PR is focused on organizing the function RunInternalPlugin() into a more easy
to understand and read function within the scorecard logic.  

**Motivation for the change:**

The intention was NOT
to change the logic but just organize it better into manageable chucks/sub-functions
instead of the single large function.  More refactoring of this section of scorecard 
code is in upcoming work that will greatly change this code further.

### CODE REVIEW

File: internal/scorecard/plugins/plugin_runner.go [hunk 1]
```diff
@@ -91,134 +92,47 @@ func RunInternalPlugin(pluginType PluginType, config BasicAndOLMPluginConfig, lo
...
-	// olm api (CSVs)
-	if err := olmapiv1alpha1.AddToScheme(scheme); err != nil {
-		return scapiv1alpha1.ScorecardOutput{}, fmt.Errorf("failed to add failed to add oml api scheme (CSVs) to client: (%v)", err)
-	}
-	dynamicDecoder = serializer.NewCodecFactory(scheme).UniversalDeserializer()
-	// if a user creates a new CRD, we need to be able to reset the rest mapper
-	// temporary kubeclient to get a cached discovery
-	kubeclient, err := kubernetes.NewForConfig(kubeconfig)
+
+	err = setupRuntimeClient()
```
Comment (joelanford) [inline internal/scorecard/plugins/plugin_runner.go]: Minor nit to decrease LoC and limit this `err` to the scope of the conditional block:

```go
if err := setupRuntimeClient(); err != nil {
	return scapiv1alpha1.ScorecardOutput{}, err
}

**Comment by joelanford** [comment]
Also, would it be possible to add some unit tests to bump that code coverage number back up?

File: internal/scorecard/plugins/plugin_runner.go (hunk 1 above)
Comment (jmccormick2001) [reply internal/scorecard/plugins/plugin_runner.go]: fixed this one.

**Comment by jmccormick2001** [comment]
> Also, would it be possible to add some unit tests to bump that code coverage number back up?

I added a new unit test, however the coveralls still reports 0.04% decreased coverage so not sure what is going on there.

**Pull request merged**